chalice/deploy/appgraph.py

```diff
@@ -38,12 +42,78 @@ def build(self, config, stage_name):
             rest_api = self._create_rest_api_model(
                 config, deployment, stage_name)
             resources.append(rest_api)
+            if config.rest_api_domain_name:
```

Comment:
@jamesls Thanks for suggesting it! I've tried to implement it in my last commit, you can check the code for it.
And what we have right now after changes:
If it will be a part of rest/websocket api, then it will be missing in resources list that are passing in `PlanStage.execute` method calling here:
 https://github.com/aws/chalice/blob/master/chalice/deploy/deployer.py#L364

In that case I need to move all plan instructions that are related to custom_domain_name and base_path_mapping from their own methods:
- https://github.com/aws/chalice/blob/master/chalice/deploy/deployer.py#L364
- https://github.com/aws/chalice/pull/1255/files/53398fe2330d09765de1b66fa2473ae8f25d04fd#diff-a03d71a10f80326ff2639fe46168d4d9R181
 to appropriate methods for restapi and for websocket api.
In that case there can be some mess with it and the duplication of code. 
And the length of that methods will increase, I think it's a bad sign. 
Maybe I miss something in existing implementation and there is a way to avoid it or maybe it will be better to left DomainName and ApiMapping (old name - BasePathMapping) as separate resources.
What do you think about it ?